src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,31 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  selectedTodo: Todo,
+  setSelectedTodo: (todo: Todo | null) => void
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo, setSelectedTodo,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+  const {
+    id, title, completed, userId,
+  } = selectedTodo;
+
+  useEffect(() => {
+    getUser(userId).then(foundUser => setUser(foundUser));
+  }, []);
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
+      {!user || !selectedTodo ? (
```

Comment:
```suggestion
      {!user ? (
```

Reply:
you don't show this component if there is no selectedTodo
![image](https://user-images.githubusercontent.com/98764641/233297614-09da750e-de75-4dee-b1d5-33c3e887e938.png)